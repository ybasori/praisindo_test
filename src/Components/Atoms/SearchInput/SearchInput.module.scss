.container {
  border: 1px solid #000;
  display: flex;
  align-items: center;
  padding: 8px 0px;
}

.input {
  border: none;
  box-sizing: border-box;
  width: 100%;
  font-size: 16px;
  &:focus-visible {
    outline: 0px;
  }
}

.btn {
  border: none;
  background-color: transparent;
  cursor: pointer;
}
