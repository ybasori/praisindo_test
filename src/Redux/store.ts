import { configureStore } from "@reduxjs/toolkit";
import nyTimesReducer from "./NyTimes/NyTimes.reducer";

export const store = configureStore({
  reducer: {
    nyTimes: nyTimesReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false }),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
