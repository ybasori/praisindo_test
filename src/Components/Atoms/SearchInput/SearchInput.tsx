import React, { useState } from "react";
import styles from "./SearchInput.module.scss";
import { AppDispatch } from "../../../Redux/store";
import { useDispatch } from "react-redux";
import { setKeyword } from "../../../Redux/NyTimes/NyTimes.reducer";

const SearchInput = () => {
  const dispatch: AppDispatch = useDispatch();
  const [form, setForm] = useState({ keyword: "" });

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    dispatch(setKeyword(form.keyword));
  };
  return (
    <form className={styles["container"]} onSubmit={onSubmit}>
      <input
        autoComplete="off"
        className={styles["input"]}
        value={form.keyword}
        name="keyword"
        onChange={(e) =>
          setForm({ ...form, [e.currentTarget.name]: e.currentTarget.value })
        }
      />
      <button className={styles["btn"]}>
        <i className="fas fa-search"></i>
      </button>
    </form>
  );
};

export default SearchInput;
