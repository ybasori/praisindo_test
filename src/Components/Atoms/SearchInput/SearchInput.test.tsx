import { render } from "@testing-library/react";
import SearchInput from "./SearchInput";
import { Provider } from "react-redux";
import { store } from "../../../Redux/store";

jest.mock("axios", () => ({
  data: { response: { docs: [] } },
}));

describe("Test SearchInput", () => {
  it("Snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <SearchInput />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
