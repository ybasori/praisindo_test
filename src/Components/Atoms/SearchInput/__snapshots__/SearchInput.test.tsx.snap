// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchInput Snapshot 1`] = `
<div>
  <form
    class="container"
  >
    <input
      autocomplete="off"
      class="input"
      name="keyword"
      value=""
    />
    <button
      class="btn"
    >
      <i
        class="fas fa-search"
      />
    </button>
  </form>
</div>
`;
